src/reanimated2/frameCallback/FrameCallbackRegistry.ts

```diff
@@ -1,54 +1,107 @@
-export default class FrameCallbackRegistry {
-  private nextCallbackId = 0;
-  private frameCallbackRegistry = new Map<number, () => void>();
-  private frameCallbackActive = new Set<number>();
-  private isFrameCallbackRunning = false;
-
-  private runAnimation() {
-    const loop = () => {
-      this.frameCallbackActive.forEach((key) => {
-        const callback = this.frameCallbackRegistry.get(key);
-        callback!();
-      });
-
-      if (this.frameCallbackActive.size > 0) {
+import { runOnUI } from '../';
+
+export interface FrameCallbackRegistryUI {
```

Comment:
Maybe we should split this file into `FrameCallbackRegistryJS.ts` and `FrameCallbackRegistryUI.ts`?